src/main/scala/com/amazon/deequ/constraints/Constraint.scala

```diff
@@ -414,7 +433,13 @@ object Constraint {
     val constraint = AnalysisBasedConstraint[MaxState, Double, Double](maxLength, assertion,
       hint = hint)
 
-    new NamedConstraint(constraint, s"MaxLengthConstraint($maxLength)")
+    val sparkAssertion = org.apache.spark.sql.functions.udf(assertion)
+
+    new RowLevelAssertedConstraint(
+      constraint,
+      s"MaxLengthConstraint($maxLength)",
+      s"StringLength-$column",
```

Comment:
Minor: Would `ColumnLength` be a better name? It remains consistent with the terminology in Glue. `StringLength` is a new term.